blocks.go

```diff
@@ -266,13 +267,76 @@ func (portal *Portal) SlackBlocksToMatrix(blocks slack.Blocks, attachments []sla
 
 	htmlText.WriteString(portal.blocksToHtml(blocks, false, userTeam))
 
+	if len(attachments) > 0 && htmlText.String() != "" {
+		htmlText.WriteString("<br>")
+	}
+
 	for _, attachment := range attachments {
 		if attachment.IsMsgUnfurl {
 			for _, message_block := range attachment.MessageBlocks {
 				renderedAttachment := portal.blocksToHtml(message_block.Message.Blocks, true, userTeam)
 				htmlText.WriteString(fmt.Sprintf("<blockquote><b>%s</b><br>%s<a href=\"%s\"><i>%s</i></a><br></blockquote>",
 					attachment.AuthorName, renderedAttachment, attachment.FromURL, attachment.Footer))
 			}
+		} else {
+			if len(attachment.Pretext) > 0 {
+				htmlText.WriteString(fmt.Sprintf("%s<br>", portal.mrkdwnToMatrixHtml(attachment.Pretext)))
+			}
+			htmlText.WriteString("<blockquote>")
+			if len(attachment.AuthorName) > 0 {
+				if len(attachment.AuthorLink) > 0 {
+					htmlText.WriteString(fmt.Sprintf("<b><a href=\"%s\">%s</a></b><br>",
+						attachment.AuthorLink, attachment.AuthorName))
+				} else {
+					htmlText.WriteString(fmt.Sprintf("<b>%s</b><br>", attachment.AuthorName))
+				}
+			}
+			if len(attachment.Title) > 0 {
+				if len(attachment.TitleLink) > 0 {
+					htmlText.WriteString(fmt.Sprintf("<strong><a href=\"%s\">%s</a></strong><br>",
+						attachment.TitleLink, portal.mrkdwnToMatrixHtml(attachment.Title)))
+				} else {
+					htmlText.WriteString(fmt.Sprintf("<strong>%s</strong><br>", portal.mrkdwnToMatrixHtml(attachment.Title)))
+				}
+			}
+			if len(attachment.Text) > 0 {
+				htmlText.WriteString(portal.mrkdwnToMatrixHtml(attachment.Text))
+			} else if len(attachment.Fallback) > 0 {
+				htmlText.WriteString(portal.mrkdwnToMatrixHtml(attachment.Fallback))
+			}
+			if len(attachment.Fields) > 0 {
+				htmlText.WriteString("<table>")
+				var short = false
+				for _, field := range attachment.Fields {
+					if !short {
+						htmlText.WriteString("<tr>")
+					}
+					htmlText.WriteString(fmt.Sprintf("<td><strong>%s</strong><br>%s</td>",
+						field.Title, portal.mrkdwnToMatrixHtml(field.Value)))
+					short = field.Short
+					if !short {
+						htmlText.WriteString("</tr>")
+					}
+				}
+				htmlText.WriteString("</table>")
+			}
+			if len(attachment.Footer) > 0 {
+				htmlText.WriteString(fmt.Sprintf("<sup>%s", portal.mrkdwnToMatrixHtml(attachment.Footer)))
+			}
+			if len(attachment.Ts) > 0 {
+				if len(attachment.Footer) > 0 {
+					htmlText.WriteString(" | ")
+				} else {
+					htmlText.WriteString("<sup>")
+				}
+				ts, _ := attachment.Ts.Int64()
+				t := time.Unix(ts, 0)
+				htmlText.WriteString(fmt.Sprintf("%s", t.Local().Format("Jan 02, 2006 15:04:05")))
```

Comment:
Just noticed that the docker image doesn't have the tzdata package installed, which is probably required to use the local timezone instead of UTC here.